metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/AlertRuleEvaluator.java

```diff
@@ -0,0 +1,86 @@
+
+public class AlertRuleEvaluator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AlertRuleEvaluator.class);
+  static final String METRIC_ANOMALY_ACTION_EVENT_TYPE = "MetricAnomalyViolation";
+
+  private static final String METRIC_ANOMALY_EVENT_CONDITION = "MetricAnomalyEventCondition";
+
+  private final StaticRuleEvaluator staticRuleEvaluator;
+  private final BaselineRuleEvaluator baselineRuleEvaluator;
+
+  public AlertRuleEvaluator(Config appConfig) {
+    QueryRequestHandler queryRequestHandler = new QueryRequestHandler(appConfig);
+    staticRuleEvaluator = new StaticRuleEvaluator(queryRequestHandler);
+    baselineRuleEvaluator = new BaselineRuleEvaluator(queryRequestHandler);
+  }
+
+  // used for testing with mock clients passed as parameters
+  AlertRuleEvaluator(
+      Config appConfig, AttributeServiceClient asClient, QueryServiceClient queryServiceClient) {
+    QueryRequestHandler queryRequestHandler =
+        new QueryRequestHandler(appConfig, queryServiceClient, asClient);
+    staticRuleEvaluator = new StaticRuleEvaluator(queryRequestHandler);
+    baselineRuleEvaluator = new BaselineRuleEvaluator(queryRequestHandler);
+  }
+
+  public Optional<NotificationEvent> process(AlertTask alertTask) throws IOException {
+    MetricAnomalyEventCondition metricAnomalyEventCondition;
+
+    if (alertTask.getEventConditionType().equals(METRIC_ANOMALY_EVENT_CONDITION)) {
+      try {
+        metricAnomalyEventCondition =
+            MetricAnomalyEventCondition.parseFrom(alertTask.getEventConditionValue());
+      } catch (InvalidProtocolBufferException e) {
+        LOGGER.error("Exception while parsing event condition", e);
+        return Optional.empty();
+      }
+    } else {
+      LOGGER.debug(
+          "Not processing alert task of EventConditionType: {}", alertTask.getEventConditionType());
+      return Optional.empty();
+    }
+
+    if (metricAnomalyEventCondition.getViolationConditionList().isEmpty()) {
+      LOGGER.debug(
+          "Received rule with empty violation conditions. tenantId: {}, eventConditionId: {}",
+          alertTask.getTenantId(),
+          alertTask.getEventConditionId());
+      return Optional.empty();
+    }
+
+    LOGGER.debug(
+        "Starting rule evaluation for rule Id {} start {} & end time {}",
+        alertTask.getEventConditionId(),
+        Instant.ofEpochMilli(alertTask.getLastExecutionTime()),
+        Instant.ofEpochMilli(alertTask.getCurrentExecutionTime()));
+
+    // todo handle multiple violation conditions
+    ViolationCondition violationCondition =
+        metricAnomalyEventCondition.getViolationConditionList().get(0);
+
+    switch (violationCondition.getConditionCase()) {
+      case STATIC_THRESHOLD_CONDITION:
+        return staticRuleEvaluator.evaluateRule(metricAnomalyEventCondition, alertTask);
+      case DYNAMIC_THRESHOLD_CONDITION:
+        return baselineRuleEvaluator.evaluateRule(metricAnomalyEventCondition, alertTask);
```

Comment:
Do we need a test for dynamic threshold condition? Current test is only for static conditions ig.